### PR TITLE
Remove shorthand for client flag in kubectl version

### PR DESCRIPTION
**Pull request by feihujiang**
Ref #15471

### CODE REVIEW

File: pkg/kubectl/cmd/version.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func NewCmdVersion(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 			cmdutil.CheckErr(err)
 		},
 	}
-	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/version.go]: Per [this comment](https://github.com/kubernetes/kubernetes/issues/15471#issuecomment-147500443), please leave the shorthand flag in and instead add
```
cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead")
```

**Comment by feihujiang** [comment]
@jlowdermilk I updated this PR according to your advice, please review.

**Comment by jlowdermilk** [comment]
lgtm

**Comment by eparis** [comment]
@feihujiang Would you please file an issue spf13/cobra that the deprecated shorthand still shows up in the man page?

**Comment by feihujiang** [comment]
@epais OK, I'll file it.

**Pull request merged**